enterprise/views.py

```diff
@@ -231,13 +244,15 @@ def post_course_specific_consent(self, request, course_id, consent_provided):
         if not request.user.is_authenticated():
             raise Http404
 
-        EnterpriseCourseEnrollment.objects.update_or_create(
-            enterprise_customer_user__user_id=request.user.id,
-            course_id=course_id,
-            defaults={
-                'consent_granted': consent_provided,
-            }
-        )
+        enrollment_deferred = request.POST.get('enrollment_deferred')
+        if enrollment_deferred is None:
+            EnterpriseCourseEnrollment.objects.update_or_create(
+                enterprise_customer_user__user_id=request.user.id,
+                course_id=course_id,
+                defaults={
+                    'consent_granted': consent_provided,
+                }
+            )
```

Comment:
Nit: instead of handling both standard and deferred enrollment in `post_course_specific_consent` it might be cleaner to implement a separate `post_deferred_course_specific_consent` handler and invoke it based on the presence of the `enrollment_defferred` parameter from the generic `post` handler below.